### DISCUSSION TITLE
The order of command-line syntax elements

### DISCUSSION THREAD

**Discussion by johnmapeson** [jgm/pandoc#9816] [Q&A] [answered]
Does Pandoc make any difference depending on the order of command-line syntax elements?

I have tried

```
pandoc -f markdown-smart -t html input.md -o output.htm --wrap=none
pandoc -f markdown-smart -o output.htm --wrap=none -t html input.md
pandoc -o output.htm --wrap=none -t html input.md -f markdown-smart
```

and all the three commands produce the same output, but maybe there are also cases, in which the order does matter?

**Comment by denismaier** [comment]
If you use multiple filters, the order does matter.

**Reply by johnmapeson** [reply]
Thanks, Denis. Do you know whether the order does matter in any other cases?

**Comment by denismaier** [comment]
I don't know.

**Comment by tarleb** [comment, marked as answer]
Other examples:

- `-v`/`--version`, `--list-...`, `--bash-completion`, and so on: only the first one of those is respected, all other args get ignored.
- `--pdf-engine-opt`: arguments are passed to the PDF engine in the same order in which they are given here.
- #7663

**Reply by johnmapeson** [reply]
Thanks, Albert!
